### PR TITLE
feat(dashboard_rbac): single dashboard api

### PR DESCRIPTION
**Pull request by amitmiran137**
### SUMMARY
enforce access for a single dashboard on api.py endpoint
the before state was returning 404 (Not found) as the dashboardFilter filtered out the dashboards
and after state is returning 403 as forbidden

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by amitmiran137** [comment]
@dpgaspar could you review ?

File: superset/dashboards/api.py [hunk 1]
```diff
@@ -460,6 +477,101 @@ def bulk_delete(self, **kwargs: Any) -> Response:
...
+        ),
+    )
+    @merge_response_func(
+        ModelRestApi.merge_show_label_columns, API_LABEL_COLUMNS_RIS_KEY
+    )
+    @merge_response_func(ModelRestApi.merge_show_columns, API_SHOW_COLUMNS_RIS_KEY)
+    @merge_response_func(
+        ModelRestApi.merge_description_columns, API_DESCRIPTION_COLUMNS_RIS_KEY
+    )
+    @merge_response_func(ModelRestApi.merge_show_title, API_SHOW_TITLE_RIS_KEY)
```
Comment (villebro) [inline superset/dashboards/api.py:498]: That's some stack of decorators

**Review by villebro — COMMENTED** [review]
LGTM, but I refer this to Daniel for review

File: superset/dashboards/api.py (hunk 1 above)
Comment (amitmiran137) [reply superset/dashboards/api.py]: remove all unessacary

**Review by dpgaspar — CHANGES_REQUESTED** [review]
LGTM but, I prefer not add/merge this. I'm assuming you need HTTP 403 for some reason, but your previous work was already secure. I can argue that a response with HTTP 404 is actually more secure then HTTP 403 on this case, if an attacker brute scans the API with id's he can infer what id's actually exist and may be subject to some other form of attack. By using HTTP 404, what you see is what you get, and no additional information is given besides what the user has access to.

Also code wise, the current new API on GET item, is based on this premise.

Thoughts?

**Comment by amitmiran137** [comment]
> LGTM but, I prefer not add/merge this. I'm assuming you need HTTP 403 for some reason, but your previous work was already secure. I can argue that a response with HTTP 404 is actually more secure then HTTP 403 on this case, if an attacker brute scans the API with id's he can infer what id's actually exist and may be subject to some other form of attack. By using HTTP 404, what you see is what you get, and no additional information is given besides what the user has access to.
> 
> Also code wise, the current new API on GET item, is based on this premise.
> 
> Thoughts?
> 

Ok then 
Sounds legit

**Pull request closed without merging**